### PR TITLE
Document reference driver

### PR DESCRIPTION
**Pull request by lubegasimon**

### CODE REVIEW

**Comment by jonludlam** [comment]
https://github.com/ocaml/odoc/pull/679/commits/97a873a3f760963e1438a1c108b10f4e4e59b0ff supersedes https://github.com/ocaml/odoc/pull/683/commits/8f3d9f640bdc1364c96470e7d68db2f32de3b728

File: doc/driver.md [hunk 1]
```diff
@@ -355,7 +355,154 @@ let generate_all odocl_files =
...
 
 ```ocaml env=e1
-let compiled = compile_all ();;
-let linked = link_all compiled;;
-let _ = generate_all linked;;
+let compiled = compile_all () in
+let linked = link_all compiled in
+generate_all linked
 ```
+
```
Comment (jonludlam) [inline doc/driver.md:362]: I think this section ought to be moved out of the driver.md file and placed elsewhere - we could certainly link to it from the driver though!

File: doc/dune [hunk 2]
```diff
@@ -6,7 +6,7 @@
   (glob_files *.ml*)
   (package odoc))
  (enabled_if
-  (> %{ocaml_version} 4.11))
+  (> %{ocaml_version} 4.11.2))
```
Comment (jonludlam) [inline doc/dune:9]: What's the purpose of this?

Comment (Julow) [reply doc/dune:9]: This is effectively disabling the test in CI.
Removing this commit and rebasing the PR on the current master solves the problem: https://github.com/Julow/odoc/actions/runs/906918515

Comment (jonludlam) [reply doc/dune:9]: great, thanks!

**Pull request merged**